tests/Unit/Domain/Creators/TimeDependence/Test_UniformRotationAboutZAxis.cpp

```diff
@@ -0,0 +1,252 @@
+
+    CHECK_ITERABLE_APPROX(
+        expected_block_map.inv_jacobian(
+            grid_coords_dv, initial_time + time_offset, functions_of_time),
+        block_map->inv_jacobian(grid_coords_dv, initial_time + time_offset,
+                                functions_of_time));
+    CHECK_ITERABLE_APPROX(
+        expected_block_map.inv_jacobian(
+            grid_coords_double, initial_time + time_offset, functions_of_time),
+        block_map->inv_jacobian(grid_coords_double, initial_time + time_offset,
+                                functions_of_time));
+
+    CHECK_ITERABLE_APPROX(
+        expected_block_map.jacobian(grid_coords_dv, initial_time + time_offset,
+                                    functions_of_time),
+        block_map->jacobian(grid_coords_dv, initial_time + time_offset,
+                            functions_of_time));
+    CHECK_ITERABLE_APPROX(
+        expected_block_map.jacobian(
+            grid_coords_double, initial_time + time_offset, functions_of_time),
+        block_map->jacobian(grid_coords_double, initial_time + time_offset,
+                            functions_of_time));
+  }
+}
+
+void test_equivalence() noexcept {
+  {
+    UniformRotationAboutZAxis<1> ur0{1.0, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<1> ur1{1.2, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<1> ur2{1.0, 3.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<1> ur3{1.0, 2.0, "RotationAngleTheta"};
+    CHECK(ur0 == ur0);
+    CHECK_FALSE(ur0 != ur0);
+    CHECK(ur0 != ur1);
+    CHECK_FALSE(ur0 == ur1);
+    CHECK(ur0 != ur2);
+    CHECK_FALSE(ur0 == ur2);
+    CHECK(ur0 != ur3);
+    CHECK_FALSE(ur0 == ur3);
+  }
+  {
+    UniformRotationAboutZAxis<2> ur0{1.0, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<2> ur1{1.2, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<2> ur2{1.0, 3.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<2> ur3{1.0, 2.0, "RotationAngleTheta"};
+    CHECK(ur0 == ur0);
+    CHECK_FALSE(ur0 != ur0);
+    CHECK(ur0 != ur1);
+    CHECK_FALSE(ur0 == ur1);
+    CHECK(ur0 != ur2);
+    CHECK_FALSE(ur0 == ur2);
+    CHECK(ur0 != ur3);
+    CHECK_FALSE(ur0 == ur3);
+  }
+  {
+    UniformRotationAboutZAxis<3> ur0{1.0, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<3> ur1{1.2, 2.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<3> ur2{1.0, 3.0, "RotationAnglePhi"};
+    UniformRotationAboutZAxis<3> ur3{1.0, 2.0, "RotationAngleTheta"};
+    CHECK(ur0 == ur0);
+    CHECK_FALSE(ur0 != ur0);
+    CHECK(ur0 != ur1);
+    CHECK_FALSE(ur0 == ur1);
+    CHECK(ur0 != ur2);
+    CHECK_FALSE(ur0 == ur2);
+    CHECK(ur0 != ur3);
+    CHECK_FALSE(ur0 == ur3);
+  }
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.Domain.Creators.TimeDependence.UniformRotationAboutZAxis",
+    "[Domain][Unit]") {
+  const double initial_time = 1.3;
+  constexpr double angular_velocity = 2.4;
+  const std::string f_of_t_name{"RotationAngle"};
+  {
+    // 2d
```

Comment:
since you're not testing the 1d case it seems to me that adding the static assert and not instantiating the class in 1d is the right approach